daft/execution/physical_plan.py

```diff
@@ -271,6 +272,131 @@ def join(
                 return
 
 
+def broadcast_join(
+    broadcaster_plan: InProgressPhysicalPlan[PartitionT],
+    reciever_plan: InProgressPhysicalPlan[PartitionT],
+    left_on: ExpressionsProjection,
+    right_on: ExpressionsProjection,
+    how: JoinType,
+    is_swapped: bool,
+) -> InProgressPhysicalPlan[PartitionT]:
+    """Broadcast join all partitions from the broadcaster child plan to each partition in the receiver child plan."""
+
+    def _emit_join_step(
```

Comment:
Does this have to be a nested function, or can we have a separate helper? The nesting makes it a bit harder to reason about which variables (e.g. `is_swapped`) are implicitly being passed through

Also a nit on the naming: the function doesn't actually emit the step, so maybe `create_join_step()` is better?